app/models/term.rb

```diff
@@ -26,6 +26,26 @@ def tests
     assignments.where(assignment_type: "test")
   end
 
+  def accepted_tasks_count
+    tasks.map{ |task| task.accepted? }.count
+  end
+
+  def notes_count
+    tasks.map{ |task| task.notes_count }.sum
+  end
+
+  def submissions_count
+    tasks.map{ |task| task.submissions_count }
```

Comment:
No, it is not OK. There's nothing about tasks in the method name. Why would other user of this method even have an idea it will return some list of numbers?